src/fideslang/validation.py

```diff
@@ -73,22 +58,22 @@ def unique_items_in_list(values: List) -> List:
     return values
 
 
-def no_self_reference(value: FidesKey, values: Dict) -> FidesKey:
+def no_self_reference(value: FidesKey, values: ValidationInfo) -> FidesKey:
     """
     Check to make sure that the fides_key doesn't match other fides_key
     references within an object.
 
     i.e. DataCategory.parent_key != DataCategory.fides_key
     """
-    fides_key = FidesKey.validate(values.get("fides_key", ""))
+    fides_key = validate_fides_key(values.data.get("fides_key", ""))
     if value == fides_key:
-        raise FidesValidationError("FidesKey can not self-reference!")
+        raise FidesValidationError("FidesKey cannot self-reference!")
     return value
 
 
 def deprecated_version_later_than_added(
-    version_deprecated: Optional[FidesVersion], values: Dict
-) -> Optional[FidesVersion]:
+    version_deprecated: Optional[str], values: ValidationInfo
```

Comment:
This is partially fixing an original bug.  `version_deprecated` was never an instance of a FidesVersion here, it was always a string.  In this function, I am converting strings to Versions here so I can properly compare them.